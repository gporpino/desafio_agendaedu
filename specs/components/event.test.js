import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { Image } from 'react-native';

import Event from '../../src/components/event';

import { FlatList } from 'react-native-gesture-handler';

const eventWithImage = {
  id: 1,
  title: 'Event 1',
  description:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque pretium nulla non arcu aliquam rhoncus eu sed leo. Aenean cursus nibh sit amet fringilla sodales. Vestibulum faucibus venenatis tempor. Morbi placerat ac massa id ultricies.',

  image:
    'https://s3-us-west-2.amazonaws.com/agendaedu-dev/schools/c5c1a933-cdef-4c9b-8a87-490f25c2538d/events/5380/attachments/1550866911-$1-original-poster-agendakids.jpeg',
  startAt: '2019-07-22T19:01:33.476Z'
};

const eventWithoutImage = {
  id: 1,
  title: 'Event 1',
  description:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque pretium nulla non arcu aliquam rhoncus eu sed leo. Aenean cursus nibh sit amet fringilla sodales. Vestibulum faucibus venenatis tempor. Morbi placerat ac massa id ultricies.',

  startAt: '2019-07-22T19:01:33.476Z'
};

describe('<Event />', () => {
  it('render with image', () => {
    const wrapper = shallow(<Event item={eventWithImage} />);

    expect(wrapper.find(Image)).toHaveLength(1);

    expect(wrapper).toMatchSnapshot();
  });

  it('render with image', () => {
    const wrapper = shallow(<Event item={eventWithoutImage} />);

    expect(wrapper.find(Image)).toHaveLength(0);

    expect(wrapper).toMatchSnapshot();
  });
});
