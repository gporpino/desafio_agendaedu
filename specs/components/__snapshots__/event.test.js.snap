// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Event /> render with image 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
>
  <Component
    style={
      Object {
        "backgroundColor": "#fff",
        "borderColor": "#fff",
        "borderRadius": 5,
        "borderWidth": 1,
        "flexDirection": "row",
        "marginBottom": 5,
        "marginLeft": 24,
        "marginRight": 24,
        "marginTop": 5,
        "shadowColor": "#733DBE",
        "shadowOffset": Object {
          "height": 1,
          "width": 1,
        },
        "shadowOpacity": 0.1,
      }
    }
  >
    <Component
      style={
        Object {
          "backgroundColor": "#733DBE",
          "borderBottomLeftRadius": 5,
          "borderTopLeftRadius": 5,
          "width": 4,
        }
      }
    />
    <Component
      source={
        Object {
          "uri": "https://s3-us-west-2.amazonaws.com/agendaedu-dev/schools/c5c1a933-cdef-4c9b-8a87-490f25c2538d/events/5380/attachments/1550866911-$1-original-poster-agendakids.jpeg",
        }
      }
      style={
        Object {
          "borderRadius": 10,
          "height": 92,
          "marginLeft": 14,
          "marginTop": 16,
          "width": 66,
        }
      }
    />
    <Component
      style={
        Object {
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
        }
      }
    >
      <Styled(Text)
        style={
          Object {
            "color": "#999999",
            "fontFamily": "sf-pro-display-medium",
            "marginBottom": 12,
          }
        }
      >
        EVENTOS
      </Styled(Text)>
      <Styled(Text)
        style={
          Object {
            "color": "#333333",
            "fontFamily": "sf-pro-display-medium",
            "fontSize": 16,
            "marginBottom": 8,
          }
        }
      >
        Event 1
      </Styled(Text)>
      <TimeBox
        size={14}
        time="2019-07-22T19:01:33.476Z"
      />
      <Styled(Text)
        style={
          Object {
            "color": "#999999",
            "fontFamily": "sf-pro-display-medium",
            "fontSize": 12,
          }
        }
      >
        Segunda, 22 de Julho às 16:01h
      </Styled(Text)>
    </Component>
  </Component>
</TouchableOpacity>
`;

exports[`<Event /> render with image 2`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
>
  <Component
    style={
      Object {
        "backgroundColor": "#fff",
        "borderColor": "#fff",
        "borderRadius": 5,
        "borderWidth": 1,
        "flexDirection": "row",
        "marginBottom": 5,
        "marginLeft": 24,
        "marginRight": 24,
        "marginTop": 5,
        "shadowColor": "#733DBE",
        "shadowOffset": Object {
          "height": 1,
          "width": 1,
        },
        "shadowOpacity": 0.1,
      }
    }
  >
    <Component
      style={
        Object {
          "backgroundColor": "#733DBE",
          "borderBottomLeftRadius": 5,
          "borderTopLeftRadius": 5,
          "width": 4,
        }
      }
    />
    <Component
      style={
        Object {
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
        }
      }
    >
      <Styled(Text)
        style={
          Object {
            "color": "#999999",
            "fontFamily": "sf-pro-display-medium",
            "marginBottom": 12,
          }
        }
      >
        EVENTOS
      </Styled(Text)>
      <Styled(Text)
        style={
          Object {
            "color": "#333333",
            "fontFamily": "sf-pro-display-medium",
            "fontSize": 16,
            "marginBottom": 8,
          }
        }
      >
        Event 1
      </Styled(Text)>
      <TimeBox
        size={14}
        time="2019-07-22T19:01:33.476Z"
      />
      <Styled(Text)
        style={
          Object {
            "color": "#999999",
            "fontFamily": "sf-pro-display-medium",
            "fontSize": 12,
          }
        }
      >
        Segunda, 22 de Julho às 16:01h
      </Styled(Text)>
    </Component>
  </Component>
</TouchableOpacity>
`;
